src/hooks/useOnboardingMessages/index.ts

```diff
@@ -0,0 +1,11 @@
+import {useMemo} from 'react';
+import useLocalize from '@hooks/useLocalize';
+import {getOnboardingMessages} from '@libs/actions/Welcome/OnboardingFlow';
+
+export default function useOnboardingMessages() {
+    const {preferredLocale} = useLocalize();
+    // Adding preferredLocale as dependency because getOnboardingMessages depends on the current locale
```

Comment:
This is an anti-pattern. More context in this proposal: https://expensify.slack.com/archives/C05LX9D6E07/p1748457444774569

Rather than using `translateLocal` in `getOnboardingMessages`, use `translate` and pass the locale in from `useLocalize`. If it needs to be used from a non-React context, you can make `locale` optional in `getOnboardingMessages`, and if it's not passed, then get the current locale from `TranslationStore.getCurrentLocale`.

That way, the component will re-render and onboarding messages will re-compute when locale changes, without having to disable any lint rules.